BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m

```diff
@@ -64,14 +64,17 @@ - (void)performAuthorizationCodeGrantWithReceivedURL:(NSURL *)URL withCompletion
     NSString *authorizationCode = [URLQueryParams valueForKey:BOXAuthURLParameterAuthorizationCodeKey];
     NSString *authorizationError = [URLQueryParams valueForKey:BOXAuthURLParameterErrorCodeKey];
 
-    if ((_nonce || serverNonce) && [serverNonce isEqualToString:_nonce] == NO) {
-        NSError *error = [[NSError alloc] initWithDomain:BOXContentSDKErrorDomain 
-                                                    code:BOXContentSDKAuthErrorAccessTokenNonceMismatch
-                                                userInfo:nil];
-        if (block) {
-            block(nil, error);
+    NSString *authenticationRedirectURIScheme = [[NSURL URLWithString:self.redirectURIString] scheme];
```

Comment:
Ah, good catch.

So if this is the check to see if we SHOULD/NEED TO do a nonce-comparison, is it better (more inclusive) to check if the server response included the 'state' query parameter (not if there is a value, if the key is there)?
We can double check if that's the better check and if it is, either do it in this commit or a separate one.